### PR TITLE
fixed asscalar error on floats

### PR DESCRIPTION
**Pull request by prateeks92**
fixed error encountered on passing floats to asscalar().

done at HackIllinois 2018

### CODE REVIEW

**Comment by eric-wieser** [comment]
This fixes #4701, I think

**Comment by eric-wieser** [comment]
Duplicates #10256, #5126

File: numpy/lib/tests/test_type_check.py [hunk 1]
```diff
@@ -442,5 +442,16 @@ def test_asfarray(self):
             asfarray, np.array([1, 2, 3]), dtype=np.array(1.0))
 
 
+class TestAsScalar(object):
+
+    def test_basic(self):
+        a = asscalar(np.array([1]))
```
Comment (mhvk) [inline numpy/lib/tests/test_type_check.py]: Use `np.asscalar`

File: numpy/lib/tests/test_type_check.py [hunk 2]
```diff
@@ -442,5 +442,16 @@ def test_asfarray(self):
...
+        assert_equal(a.__class__, int)
+        assert_raises(ValueError, asscalar,[1,2])
```
Comment (mhvk) [inline numpy/lib/tests/test_type_check.py]: Please add spaces after `,` (pep8)

**Review by mhvk — COMMENTED** [review]
See https://github.com/numpy/numpy/pull/5126#issuecomment-74782471 where it was suggested to just leave things as is. If we do that, we should at least close the issue! Though I think we might as well just apply this fix, and add the test (see in-line comments).

**Pull request closed without merging**